lib/queue.rb

```diff
@@ -1,31 +1,102 @@
 class Queue
+# set queue_size here to something like 20 or whatever
+queue_size = 20
```

Comment:
This should be a constant, the fact that it's not gives you scope problems and failing tests